### ISSUE TITLE
Instructor Question

### ISSUE THREAD

**Issue by nswalters** [nswalters/bear-watcher#37]
@luketlancaster @michaelclark2

# Question:

When creating cards with specific row counts, is there a better way to handle added the proper divs/classes for rows where required?

Since I only wanted to added a row for every two timestamps, it was relatively easy using modulo checks against the index counts to inject a row div where required.  However, this feels like it's not the most effective way of doing this.

```js
const addTimestampData = (timestamps) => {

  let domString = '';

  for (let index = 0; index < timestamps.length; index++) {

    // Do some timestamp mangling to convert to human-readable timestamp
    const timestamp = new Date(timestamps[index]);

    const convertedTimestamp = timestamp.getFullYear() + '-' +
      ('0' + (timestamp.getMonth() + 1)).slice(-2) + '-' +
      ('0' + timestamp.getDate()).slice(-2) + ' ' +
      timestamp.getHours() + ':' +
      ('0' + timestamp.getMinutes()).slice(-2) + ':' +
      ('0' + timestamp.getSeconds()).slice(-2);

    // Start 'row' div every two timestamps
    if (index % 2 === 0) {
      domString += `<div class="row">`;
    }

    // Add timestamp content to row
    domString += `
    <div class="col col-6 px-0">
      <div>${convertedTimestamp}</div>
    </div>
    `

    // End 'row' div after every second element (every index that's odd)
    if (index % 2 !== 0) {
      domString += `</div>`
    }
  };

  // Handles adding the closing row div if an odd number of timestamps are present
  if (timestamps.length % 2 !== 0) {
    domString += `</div>`;
  }

  return domString;
}
```

**Comment by nswalters** [comment]
i.e. I'm using the above to create the card-body sections with the timestamps in two column rows as shown in this example: 

![Bear card layout with live times](https://user-images.githubusercontent.com/10491407/83370287-8e803980-a384-11ea-81d2-bf8cca89ab95.png)

**Comment by luketlancaster** [comment]
🤔 Instead of using `row` could you use the bootstrap `col` classes? Or even you could go anti-bootstrap and style it yourself with flex...making it so there's two items, then it breaks to a new line, and give them a `border-bottom`? Though I might be misunderstanding the goal.

Also check out [moment.js](https://momentjs.com/) for all your Date Parsing/Displaying needs...

**Comment by nswalters** [comment]
Thanks @luketlancaster.  I've avoided any imports as we hadn't covered any of that yet. :)

The end goal is to have two columns of timestamps on each side of the card (successful and missed).

As I understood Bootstrap flexbox grid, the required hierarchy is `container > row > col` in order to get the columns placed properly.  I remember before that I was having issues with configuring  columns with only a specific number of columns that were also responsive when re-sized.  I'll do some more playing around with flex and see what I can come up with.

**Comment by luketlancaster** [comment]
yeah you wouldn't be able to import moment, but you could use a cdn for it!

And that's the hierarchy, yeah. But each row is 12 columns, and if two containers add up to > 12 cols, than the second will flow onto the next line (still in the same "row" container). So if your timestamp divs were `col-6` or something, than there should be two per row. I say "should" because who even understands the dark magic that is CSS
